runtime/src/iree/hal/drivers/hip/pending_queue_actions.c

```diff
@@ -46,6 +46,15 @@ typedef enum iree_hal_hip_queue_action_state_e {
   IREE_HAL_HIP_QUEUE_ACTION_STATE_ZOMBIE,
 } iree_hal_hip_queue_action_state_t;
 
+// Count actions or things with count O(number-of-action) like work items.
+// We assume the number of pending actions will be small enough so we don't
+// resort to iree_host_size_t.
+typedef int32_t action_count_t;
+// How many work items must complete in order for an action to complete.
```

Comment:
this feels sketchy - is relying on fixed counts really the best information available? if other actions are added in the future this will be harder to manage and it's not clear what this actually does from the descriptor or the code using it.

Reply:
I renamed the constant to `total_work_items_to_complete_an_action` and changed the doc. I added this constant because before it was just the number 2 used in an expression which was even more confusing.
I think this can be removed as there are more simplifications possible. We don't have to ping-pong actions between the worker and completion threads. We don't need to do that as we are not using HIP stream callbacks anymore and there are no restrictions on making HIP API calls. I left this change for another PR as this one turned out to be big enough already.

Reply:
One of the clean-ups I was planning as we move forward was moving the deletion directly onto the completion thread. It also simplifies the mental model of this quite a bit.